---
title: " TIL&TEL 20.06.08~20.06.14  "
date: "2020-06-14T01:23:45.678Z"
template: "post"
draft: false
slug: "/posts/TIL&TEL_20-06-14/"
category: "TIL&TEL"
tags:
- "TIL"
- "TEL"


description: " 20.06.08~20.06.14 배운것, 씨름한것들 "
---

## TIL( Today I Learned)

- ### 20-06-08 월

  - 프로세스 동작
  - sql 짜기
  - 프로젝트 회의

- ### 20-06-09 화

  - 에러 잡기

  - looping sleep error 잡기

    

- ### 20-06-10 수

  - 작업 프로세스 문서 정리

    

- ### 20-06-11 목

  - STT sync 맞추는 DB 처리 제대로 동작하도록
  - team work culture 논의

- ### 20-06-12 금

  - 엔진 에러 디버깅
    - 오디오 파일의 경우 일정 사이즈 이하면 오디오 재생이 거의 안되다시피 한다. 오디로 파일을 처리 하는 곳에서 에러가 발생한다.
  - isort, black, flake8 적용해보기

- ### 20-06-14 일

  - 블로그 쓰기

    - 블로그계 F class!! 
      - 어떤거야? 
      - 쓸 필요가 있어?
      - 어떻게 써? 



## TEL (Trial and Error Log)

- ### 20-06-08

  - Error:django timezone utc 기준으로 동작이 된다.
    - Situation: time_zone asia/seoul 로 해놨는데 그대로 동작이 안된다
    - Reason: USE_TZ = True 임
    - Solution: USE_TZ 삭제
  - Error: 쿼리 동작이 안됨
    - Situation: 테스트 하기 위해 query 작성해서 동작하고 있었는데 막히고 있었다.
    - Reason: 엉뚱한 DB에 접속을 하고 있었다.
    - Solution: 접속 DB 수정
  - Error: while loop 에러
    - Situation: while 문 내에서 DB에서 불러오는 값이 초기화가 안된다.
    - Reason: DB 커넥션에 들어가는 커넥션을 물고 있는 클래스 context 동작의 문제 
    - Solution: context 진입을 while 문안에서 새로 시작
  - Error: 프로세스와 django에서 공용으로 쓰는 DB의 sync가 맞지 않는다.
    - Situation: 프로세스, django 같이 하나의 row를 보고 돌고 있는데 숫자가 안맞음
    - Reason: django 에서 row 값을 불러오고 다시 수정해서 저장하는 식으로 하다보니 그 사이에 row value가 바뀌면 값이 씹히게 됨
    - Solution: django.db.F 사용 

- ### 20-06-10

  - Error: tuple parsing error

    - Situation: Tuple 파싱해서 Dict 로 변환하는데 있어서 문제가 생긴다. 
    - Reason: Tuple 내에 Empyt Tuple 이 한개 또는 그 이상 들어있어도 해당 Tuple 값이 있는 걸로 인식 한다.
    - Solution: Tuple 내에서 Loop 도는 것에도 `if` 조건절 추가함

  - Error: tuple 값 parsing 하는것 확인해서 고쳤는데 안됨

    - Situation: SQL 불러온 값 DICT 로 파싱
    - Reason: 코드 수정후 반영을 안함
    - Solution: 코드 배포

  - Error: object of type 'NoneType' has no len()

     

    - Situation: len(None)
    - Reason: None 처리에는 제한이 있다.
    - Solution: None은 is or is not 으로 처리

- ### 20-06-13

  - Error: TypeError: unhashable type: 'dict'
    - Situation: dict의 collection set 함수에 넣으니 이럼
    - Reason: hashable 하지 않은 것은 set 처리 못함
    - Solution: 단일 dict는 가능하나, dict collection에서는 불가하다.
  - Error: DB Load 한값 처리 못함
    - Situation: DB 에서 반환한 값 불러서 사용하는 줄 알고 있었는데 안되고 있음
    - Reason: db에서 불러온 값이 tuple in tuple 인데 tuple 처럼 취급
    - Solution: tuple unpacking 추가함